sys/kern/sleepq.c

```diff
@@ -324,7 +324,8 @@ int sleepq_wait_timed(void *wchan, const void *waitpt, mtx_t *mtx,
     callout_schedule(&td->td_slpcallout, timeout);
   }
 
-  td->td_flags |= (timeout > 0) ? TDF_SLPTIMED : TDF_SLPINTR;
+  td->td_flags |= TDF_SLPINTR;
```

Comment:
Yes, I did change the semantics as I think it was a bug that `TDF_SLPINTR` was not set. See discussion here: https://discord.com/channels/951841212267130950/1054133625731424337

Reply:
Okay, I agree with that, but I also think that @cahirwpz should take a look at that.

Reply:
I think it has to do with possible deficiencies in `sleepq` API. ATM we have two `wait` functions - with and without timeout. IMO there should be `wait` and `wait_intr` with optional timeout. That should let us express all use cases for putting a thread to sleep.

Please note that non-interruptible `wait` should probably be only used by the code that does not interface user-space directly - i.e. the code that does not understand the concept of a signal or need not be burdened with complexity of extra error handling (`ETIMEDOUT`). Examples would be waking up a thread from interrupt or "fast" device drivers code. On contrary the interruptible `wait` should be used in "slow" device drivers code (where `read` or `write` may be interrupted) and all syscalls that can tolerate being interrupted.

Perhaps what I wrote above should be redacted and put somewhere into `sleepq.h`.

Reply:
To answer the question - the change is ok, since function is documented as follows:

```c
/*! \brief Performs interruptible sleep with timeout.
 *
 * Timed sleep is also interruptible. [...]
 */
```